MdePkg/Library/StackCheckLib/StackCheckLibDynamicInit.c

```diff
@@ -0,0 +1,44 @@
+/** @file
+  Provides the required functionality for initializing
+  the stack cookie value.
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#include <Base.h>
+#include <Uefi.h>
+
+#include <Library/RngLib.h>
+
+#if defined (__GNUC__) || defined (__clang__)
+extern VOID  *__stack_chk_guard;
+#elif defined (_MSC_VER)
+extern VOID  *__security_cookie;
+#endif
+
+/**
+  Initialize the security cookie.
+**/
+NO_STACK_COOKIE
+EFI_STATUS
+EFIAPI
+InitializeSecurityCookie (
+  VOID
+  )
+{
+  // Only generate a random value for MSVC builds. GCC builds which use
+  // -fstack-protector-strong or -fstack-protector-all may encounter a
+  // stack overflow with a random cookie value. The exact cause of this
+  // behavior needs to be investigated further.
+ #ifdef _MSC_VER

```

Comment:
Any progress on this issue?  Should this be moved to a bug report and update the comment to state that MSVC uses random number and GCC uses a static value?

Also, the INF syntax supports a tool chain family filter and this logic can be moved into different source files for MSFT and GCC and remove the use of #ifdef _MSC_VER

Reply:
Per my other comment it looks like we can define `NO_STACK_COOKIE` to be empty in Base.h I wonder if that is related. I know for clang we have a dynamic cookie and no issue, but we don't depend on NO_STACK_COOKIE

Reply:
I think clang might make using MSFT/GCC a little more complex. I think there is a non zero chance that clang might do it both ways depending on which one you use. The clang that makes PE/COFF and PDB likely uses the MSABI form, and some of the other clangs likely use the GCC form? So we should no assume that is just going to work. You can also cheat and move the global to assembly and use both names.

Reply:
Same on this one, I will follow up on this one when I return

Reply:
Resolving as the dynamic instance is dropped for now.